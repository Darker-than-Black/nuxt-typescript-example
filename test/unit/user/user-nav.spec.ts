import { RouterLinkStub } from '@vue/test-utils'
import VueTestInstance from '~/test/factory/vue-instance'
import UserNav from '~/components/user/user-nav.vue'

describe('user-nav.vue', () => {
  const wrapper = VueTestInstance(UserNav, {
    options: {
      stubs: { NuxtLink: RouterLinkStub }
    }
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('check emit "logout"', () => {
    wrapper.find('button.btn').trigger('click')
    expect(wrapper.emitted().logout).toBeTruthy()
  })
})
