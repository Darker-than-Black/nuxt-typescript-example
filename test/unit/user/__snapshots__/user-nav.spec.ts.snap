// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user-nav.vue should match the snapshot 1`] = `
<div>
  <div class="d-flex align-items-center user__row">
    <div class="user__icon">
      VD
    </div> <span class="user__name">
      Влад Данилюк
    </span>
  </div>
  <ul class="user-nav__list">
    <li class="link user-nav__item">
      Налаштування
      <img src="/icons/Settings.svg" alt="settings" class="img"></li>
  </ul>
  <div class="user__row"><button type="button" class="btn btn-gray">
      Вихід
      <img src="/icons/Exit.svg" alt="Exit" class="img"></button></div>
</div>
`;
